pkg/cloud_provider/file/file.go

```diff
@@ -189,13 +209,46 @@ func (manager *gcfsServiceManager) DeleteInstance(ctx context.Context, obj *Inst
 	}
 
 	glog.Infof("Starting DeleteInstance cloud operation")
-	_, err = manager.instancesService.Delete(instanceURI(obj.Project, obj.Location, obj.Name)).Context(ctx).Do()
+	op, err := manager.instancesService.Delete(instanceURI(obj.Project, obj.Location, obj.Name)).Context(ctx).Do()
 	if err != nil {
 		return fmt.Errorf("DeleteInstance operation failed: %v", err)
 	}
 
-	// Always return error and check for instance in the subsequent calls
-	return fmt.Errorf("DeleteInstance operation started")
+	err = manager.waitForOp(ctx, op)
+	if err != nil {
+		return fmt.Errorf("WaitFor DeleteInstance operation failed: %v", err)
+	}
+
+	instance, err = manager.GetInstance(ctx, obj)
```

Comment:
not sure I like that `!isNotFoundErr(err)` is hidden inside `GetInstance` and it simply doesn't return an instance instead of throwing an error. Technically it might be better but it also goes against the pattern that we see in all of Kubernetes so to me it was fairly confusing

Reply:
done